plugins/hotkeyshortcuts.koplugin/main.lua

```diff
@@ -0,0 +1,581 @@
+    end
+
+    local ReaderWikipedia = require("apps/reader/modules/readerwikipedia")
+    ReaderWikipedia.registerKeyEvents = function(readerwikipedia)
+    end
+
+    local ReaderUI = require("apps/reader/readerui")
+    ReaderUI.registerKeyEvents = function(readerui)
+        if Device:hasKeys() then
+            readerui.key_events.Home = { { "Home" } }
+            if Device:hasDPad() and Device:useDPadAsActionKeys() then
+                readerui.key_events.KeyContentSelection = { { { "Up", "Down" } }, event = "StartHighlightIndicator" }
+            elseif Device:hasKeyboard() then
+                readerui.key_events.Reload = { { "F5" } }
+            end
+        end
+    end
+
+    local FileManager = require("apps/filemanager/filemanager")
+    local FileChooser = require("ui/widget/filechooser")
+    FileManager.registerKeyEvents = function(filemanager)
+        if Device:hasKeys() then
+            filemanager.key_events.Home = { { "Home" } }
+            -- Ensure file_chooser is initialized before accessing it
+            if not filemanager.file_chooser then
+                filemanager.file_chooser = FileChooser:new()
+            end
+            -- Override the menu.lua way of handling the back key
+            filemanager.file_chooser.key_events.Back = { { Device.input.group.Back } }
+            if not Device:hasFewKeys() then
+                -- Also remove the handler assigned to the "Back" key by menu.lua
+                filemanager.file_chooser.key_events.Close = nil
+            end
+        end
+    end
+
+    local FileSearcher = require("apps/filemanager/filemanagerfilesearcher")
+    FileSearcher.registerKeyEvents = function(filemanagerfilesearcher)
+        if Device:hasKeyboard() then
+            filemanagerfilesearcher.key_events.ShowFileSearchBlank = {
+                { "Alt", "Shift", "F" }, { "Ctrl", "Shift", "F" },
+                event = "ShowFileSearch",
+                args = ""
+            }
+        end
+    end
+
+    local FileManagerMenu = require("apps/filemanager/filemanagermenu")
+    FileManagerMenu.registerKeyEvents = function(filemanagermenu)
+        if Device:hasKeys() then
+            filemanagermenu.key_events.ShowMenu = { { "Menu" } }
+        end
+    end
+end -- overrideConflictingFunctions()
+
+--[[
+    This function checks if the `settings_data` exists and if it has been marked as updated.
+    If both conditions are met, it flushes the `settings_data` and resets the `updated` flag to false.
+--]]
+function HotKeyShortcuts:onFlushSettings()
+    if self.settings_data and self.updated then
+        self.settings_data:flush()
+        self.updated = false
+    end
+end
+
+function HotKeyShortcuts:updateProfiles(action_old_name, action_new_name)
+    for _, section in ipairs({ "hotkeyshortcuts_fm", "hotkeyshortcuts_reader" }) do
+       local hotkeyshortcuts = self.settings_data.data[section]
+        for shortcut_name, shortcut in pairs(hotkeyshortcuts) do
```

Comment:
Not familiar with profiles, so it probably needs a quick reading by @hius07 (even if probably copy & pasted from the gestures plugin.)

Reply:
yes, no changes here.